internal/manifest/cr_deletion_check.go

```diff
@@ -0,0 +1,48 @@
+package manifest
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	declarative "github.com/kyma-project/lifecycle-manager/internal/declarative/v2"
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewCustomResourceDeletionCheck creates a check that verifies that the Resource CR in the remote cluster is deleted.
+func NewCustomResourceDeletionCheck() *CRDeletionCheck {
+	return &CRDeletionCheck{}
+}
+
+type CRDeletionCheck struct{}
+
+func (c *CRDeletionCheck) Run(
+	ctx context.Context,
+	clnt client.Client,
+	obj declarative.Object,
+) (bool, error) {
+	manifest, ok := obj.(*v1beta2.Manifest)
+	if !ok {
+		return false, v1beta2.ErrTypeAssertManifest
+	}
+	if manifest.Spec.Resource == nil {
+		return true, nil
+	}
+
+	name := manifest.Spec.Resource.GetName()
+	namespace := manifest.Spec.Resource.GetNamespace()
+
+	resourceCR := &unstructured.Unstructured{}
+	if err := clnt.Get(ctx,
+		client.ObjectKey{Name: name, Namespace: namespace}, resourceCR); err != nil {
+		if util.IsNotFound(err) {
+			return true, nil
+		}
+		return false, errors.Wrap(err, "failed to fetch default resource CR")
+	}
+
+	return false, clnt.Delete(ctx, resourceCR)
```

Comment:
delete resource cr should not happens here, LM delete it in https://github.com/kyma-project/lifecycle-manager/blob/cc7c6e49a9da791f6d90188a63b6184abbf4b819/internal/manifest/custom_resource.go#L70